Decorated tests not being picked up by pytest

With a `test.py` that looks like

```python
from testbook import testbook

@testbook('my_notebook.ipynb')
def test_foo(tb):
    assert False, "foo"

def test_bar():
    assert False, "bar"
```
running `pytest test.py` yields

```
=================================== test session starts ===================================
platform linux -- Python 3.7.9, pytest-6.0.2, py-1.9.0
collected 1 item                                                                          

test.py F                        [100%]

======================================== FAILURES =========================================
________________________________________ test_bar _________________________________________

    def test_bar():
>       assert False, "bar"
E       AssertionError: bar
E       assert False

test.py:9: AssertionError
================================= short test summary info =================================
FAILED test.py::test_bar - AssertionE...
==================================== 1 failed in 0.17s ====================================
```

Only the non-testbook test is picked up, even though they both are prefixed with `test`.

---------
On the other hand, when I use a context manager in `test.py`:
```python
from testbook import testbook

def test_foo():
    with testbook('my_notebook.ipynb') as tb:
        assert False, "foo"

def test_bar():
    assert False, "bar"
```

then *both* the tests are picked up.

```
=================================== test session starts ===================================
platform linux -- Python 3.7.9, pytest-6.0.2, py-1.9.0
collected 2 items                                                                         

test.py FF                       [100%]

======================================== FAILURES =========================================
________________________________________ test_foo _________________________________________

    def test_foo():
        with testbook('my_notebook.ipynb') as tb:
>           assert False, "foo"
E           AssertionError: foo
E           assert False

test.py:7: AssertionError
________________________________________ test_bar _________________________________________

    def test_bar():
>       assert False, "bar"
E       AssertionError: bar
E       assert False

test.py:10: AssertionError
================================= short test summary info =================================
FAILED test.py::test_foo - AssertionE...
FAILED test.py::test_bar - AssertionE...
==================================== 2 failed in 1.20s ====================================
```

--- a/testbook/testbook.py
+++ b/testbook/testbook.py
@@ -8,7 +8,7 @@ from .client import TestbookNotebookClient
 
 class testbook:
     """`testbook` acts as function decorator or a context manager.
-    
+
     When the function/with statement exits the kernels started when
     entering the function/with statement will be terminated.
 
@@ -16,7 +16,7 @@ class testbook:
     will be passed as first argument to the decorated function.
     """
     # Developer notes:
-    # 
+    #
     # To trick pytest, we mimic the API of unittest.mock.patch in testbook.
     # Notably, the following elements are added:
     # * attribute_name, Class attribute (see below)

--- a/testbook/testbook.py
+++ b/testbook/testbook.py
@@ -1,15 +1,30 @@
-import nbformat
+import functools
+from unittest.mock import DEFAULT
 
-try:
-    from pytest import fixture
-except ImportError:  # pragma: no cover
-    def fixture(func, *args, **kwargs):
-        return func
+import nbformat
 
 from .client import TestbookNotebookClient
 
 
 class testbook:
+    """`testbook` acts as function decorator or a context manager.
+    
+    When the function/with statement exits the kernels started when
+    entering the function/with statement will be terminated.
+
+    If `testbook` is used as a decorator, the `TestbookNotebookClient`
+    will be passed as first argument to the decorated function.
+    """
+    # Developer notes:
+    # 
+    # To trick pytest, we mimic the API of unittest.mock.patch in testbook.
+    # Notably, the following elements are added:
+    # * attribute_name, Class attribute (see below)
+    # * new, Instance attribute (see __init__)
+    # * patchings, wrapper attributes (see __call__)
+
+    attribute_name = None
+
     def __init__(self, nb, execute=None, timeout=60, kernel_name='python3', allow_errors=False):
         self.execute = execute
         self.client = TestbookNotebookClient(
@@ -18,6 +33,8 @@ class testbook:
             allow_errors=allow_errors,
             kernel_name=kernel_name
         )
+
+        self.new = DEFAULT
 
     def _prepare(self):
         if self.execute is True:
@@ -34,12 +51,11 @@ class testbook:
         self.client._cleanup_kernel()
 
     def __call__(self, func):
+        @functools.wraps(func)
         def wrapper(*args, **kwargs):  # pragma: no cover
             with self.client.setup_kernel():
                 self._prepare()
                 func(self.client, *args, **kwargs)
 
-        wrapper.__name__ = func.__name__
-        wrapper.__doc__ = func.__doc__
-
-        return fixture(wrapper)
+        wrapper.patchings = [self]
+        return wrapper

--- a/testbook/tests/test_testbook.py
+++ b/testbook/tests/test_testbook.py
@@ -32,6 +32,12 @@ def test_testbook_decorator_with_markers(nb, cell_index_args, expected_result):
     assert nb._cell_index(cell_index_args) == expected_result
 
 
+@pytest.mark.parametrize("cell_index_args, expected_result", [(2, 2), ('hello', 1)])
+@testbook('testbook/tests/resources/inject.ipynb', execute=True)
+def test_testbook_decorator_with_markers_order_does_not_matter(nb, cell_index_args, expected_result):
+    assert nb._cell_index(cell_index_args) == expected_result
+
+
 def test_testbook_execute_all_cells_context_manager():
     with testbook('testbook/tests/resources/inject.ipynb', execute=True) as tb:
         for cell in tb.cells[:-1]:

